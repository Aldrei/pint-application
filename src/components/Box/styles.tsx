import styled from 'styled-components'

export const BoxStyle = styled.div`
  margin: '20px'
`