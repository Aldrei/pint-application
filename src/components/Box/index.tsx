import React from 'react'

import { BoxStyle } from './styles'

export interface Props  { 
  children?: React.ReactNode
}

const Box = (props: Props): React.ReactElement => {
  return (
    <BoxStyle className='boxComp'>{props.children}</BoxStyle>
  );
}

export default Box;