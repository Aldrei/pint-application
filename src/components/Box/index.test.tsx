import React from 'react';

import { render } from '@testing-library/react';

import Box from './index';

describe('Box component', () => {
  it('Should render correctly', () => {
    const nodeEl = render(<Box><h1 className="child-test">Test</h1></Box>);

    const nodeElChild = nodeEl.baseElement.getElementsByClassName('child-test');
    expect(nodeElChild.length).toBe(1);
    
    expect(nodeEl.baseElement).toMatchSnapshot();
  });
});
