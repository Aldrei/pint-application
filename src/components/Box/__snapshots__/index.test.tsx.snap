// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component Should render correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bczRLJ imWkNv boxComp"
    >
      <h1
        class="child-test"
      >
        Test
      </h1>
    </div>
  </div>
</body>
`;
